tests/std/tests/P0784R7_library_support_for_more_constexpr_containers/test.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <assert.h>
+#include <memory>
+#include <string.h>
+#include <string>
+#include <type_traits>
+#include <utility>
+
+struct X {};
+
+template <class _Void, class Ty, class... Types>
+inline constexpr bool can_construct_at = false;
+
+template <class Ty, class... Types>
+inline constexpr bool can_construct_at<
+    std::void_t<decltype(std::construct_at(std::declval<Ty*>(), std::declval<Types>()...))>, Ty, Types...> = true;
+
+static_assert(can_construct_at<void, int>);
+static_assert(can_construct_at<void, const int>);
+static_assert(can_construct_at<void, volatile int>);
+static_assert(can_construct_at<void, const volatile int>);
+static_assert(can_construct_at<void, int, int>);
+static_assert(can_construct_at<void, const int, int>);
+static_assert(can_construct_at<void, volatile int, int>);
+static_assert(can_construct_at<void, const volatile int, int>);
+static_assert(can_construct_at<void, int, int&>);
+static_assert(can_construct_at<void, const int, int&>);
+static_assert(can_construct_at<void, volatile int, int&>);
+static_assert(can_construct_at<void, const volatile int, int&>);
+
+static_assert(!can_construct_at<void, int, X>);
+static_assert(!can_construct_at<void, X, int>);
+
+static_assert(!can_construct_at<void, int&>); // note that references can be constructed by not new'd
+static_assert(!can_construct_at<void, const int&>);
+static_assert(!can_construct_at<void, volatile int&>);
+static_assert(!can_construct_at<void, const volatile int&>);
+static_assert(!can_construct_at<void, int&, int>);
+static_assert(!can_construct_at<void, const int&, int>);
+static_assert(!can_construct_at<void, volatile int&, int>);
+static_assert(!can_construct_at<void, const volatile int&, int>);
+static_assert(!can_construct_at<void, int&, int&>);
+static_assert(!can_construct_at<void, const int&, const int&>);
+static_assert(!can_construct_at<void, volatile int&, volatile int&>);
+static_assert(!can_construct_at<void, const volatile int&, const volatile int&>);
+
+static_assert(can_construct_at<void, X>);
+static_assert(can_construct_at<void, X, X>);
+static_assert(can_construct_at<void, X, const X&>);
+static_assert(can_construct_at<void, X, X&>);
+static_assert(can_construct_at<void, X, X&&>);
+
+static_assert(can_construct_at<void, std::string>);
+static_assert(can_construct_at<void, std::string, std::size_t, char>);
+static_assert(!can_construct_at<void, std::string, std::size_t, char, char>);
+static_assert(!can_construct_at<void, std::string, X>);
+
+static_assert(noexcept(std::construct_at(std::declval<int*>(), 42)));

```

Comment:
Should you have a comment that all of these `noexcept` tests are testing our strengthenings?